fluss-client/src/main/java/com/alibaba/fluss/client/utils/ClientUtils.java

```diff
@@ -136,16 +136,13 @@ public static Long getPartitionId(
     }
 
     public static int getBucketId(
-            byte[] keyBytes,
-            InternalRow key,
-            @Nullable LakeTableBucketAssigner lakeTableBucketAssigner,
-            int numBuckets,
-            MetadataUpdater metadataUpdater) {
-        if (lakeTableBucketAssigner == null) {
-            return HashBucketAssigner.bucketForRowKey(keyBytes, numBuckets);
+            byte[] bucketKeyBytes,
+            @Nullable LakeBucketAssigner lakeBucketAssigner,
+            int numBuckets) {
+        if (lakeBucketAssigner == null) {
+            return HashBucketAssigner.bucketForRowKey(bucketKeyBytes, numBuckets);
         } else {
-            return lakeTableBucketAssigner.assignBucket(
-                    keyBytes, key, metadataUpdater.getCluster());
+            return lakeBucketAssigner.assignBucket(bucketKeyBytes);
```

Comment:
It is still not unified and complex to assign bucket. We can introduce a `BucketingFunction#bucketing(byte[] key, int numBuckets)` to unify the bucket assigned instead of using a util method and passing optional lake bucket assigner. We should hold only one not-null bucket assigner, that will be easier to use.